src/main/kotlin/no/nav/familie/ef/sak/service/JournalføringService.kt

```diff
@@ -0,0 +1,133 @@
+        behandlingService.oppdaterJournalpostIdPåBehandling(journalpostId, behandling)
+    }
+
+    private fun settSøknadPåBehandling(journalpostId: String) {
+        hentJournalpost(journalpostId).dokumenter
+                ?.filter { dokument -> DokumentBrevkode.erGyldigBrevkode(dokument.brevkode) && harOriginalDokument(dokument) }
+                ?.map { Pair(DokumentBrevkode.fraBrevkode(it.brevkode), hentDokument(journalpostId, it.dokumentInfoId)) }
+                ?.forEach {
+                    konverterTilSøknadsobjekt(it)
+                }
+    }
+
+    private fun konverterTilSøknadsobjekt(it: Pair<DokumentBrevkode, ByteArray>) {
+        try {
+            when (it.first) {
+                DokumentBrevkode.OVERGANGSSTØNAD -> {
+                    objectMapper.readValue(it.second, SøknadOvergangsstønad::class.java)
+                    // TODO: Bent må sette inn i databasen når domenemodellen er klar
+                }
+                DokumentBrevkode.BARNETILSYN -> {
+                    objectMapper.readValue(it.second, SøknadBarnetilsyn::class.java)
+                    // TODO: Bent må sette inn i databasen når domenemodellen er klar
+                }
+                DokumentBrevkode.SKOLEPENGER -> {
+                    objectMapper.readValue(it.second, SøknadSkolepenger::class.java)
+                    // TODO: Bent må sette inn i databasen når domenemodellen er klar
+                }
+            }
+        } catch (e: JsonProcessingException) {
+            secureLogger.warn("Kan ikke konvertere journalpostDokument til søknadsobjekt", e)
+            logger.warn("Kan ikke konvertere journalpostDokument til søknadsobjekt ${e.javaClass.simpleName}")
+        }
+    }
+
+    private fun harOriginalDokument(dokument: no.nav.familie.kontrakter.felles.journalpost.DokumentInfo): Boolean =
+            dokument.dokumentvarianter?.contains(Dokumentvariant(variantformat = DokumentVariantformat.ORIGINAL.toString()))
+                    ?: false
+
+    private fun oppdaterJournalpost(journalpost: Journalpost, dokumenttitler: Map<String, String>?, fagsakId: UUID) {
+        val oppdatertJournalpost = OppdaterJournalpostRequest(
+                bruker = journalpost.bruker?.let {
+                    DokarkivBruker(
+                            idType = IdType.valueOf(it.type.toString()),
+                            id = it.id
+                    )
+                },
+                tema = journalpost.tema,
+                behandlingstema = journalpost.behandlingstema,
+                tittel = journalpost.tittel,
+                journalfoerendeEnhet = journalpost.journalforendeEnhet,
+                sak = Sak(
+                        fagsakId = fagsakId.toString(),
+                        fagsaksystem = "EF",
+                        sakstype = "FAGSAK"
+                ),
+                dokumenter = dokumenttitler?.let {
```

Comment:
Litt usikker på denne, hvordan er det tenkt att den skal virke?
Hvis man ikke sender med dokumenttitler nå så settes ikke noe dokumenter? 

Er det ønskelig å istedet ha

```kotlin
journalpost.dokumenter?.map { dokumentInfo ->
            DokumentInfo(
                    dokumentInfoId = dokumentInfo.dokumentInfoId,
                    tittel = dokumenttitler?.get(dokumentInfo.dokumentInfoId) ?: dokumentInfo.tittel,
                    brevkode = dokumentInfo.brevkode
            )
        }
```

Reply:
Journalpostsystemet fungerer på følgende måte for en del av feltene:
Hvis feltet "X" er tomt så blir det ikke gjort oppdateringer/endringer av feltet X.

Sånn sett logges det ingen endring av dokumenter hvis saksbehandler ikke har gjort noen endringer.